src/App.js

```diff
@@ -1,17 +1,85 @@
 import React, { Component } from 'react';
+import axios from 'axios';
 import './App.css';
 import Board from './components/Board';
+import Status from './components/Status';
 
 class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      boardName: 'Ada-Lovelace',
+      boards: [],
+      boardsUrl: 'https://inspiration-board.herokuapp.com/boards/',
+      status: {
+        message: 'Successfully loaded the page',
+        type: 'success'
+      }
+    };
+  }
+
+  updateStatus = (message, type) => {
+    this.setState({
+      status: {
+        message: message,
+        type: type
+      }
+    });
+  }
+
+  componentDidMount() {
+    this.updateStatus('Retrieving inspirational boards', 'success');
+
+    axios.get(this.state.boardsUrl)
+      .then((response) => {
+        this.updateStatus('Successfully retrieved all boards', 'success');
+        let updatedBoards = response.data.map((boardInfo) => {
+          return boardInfo.board;
+        });
+        this.setState({
+          boards: updatedBoards
+        });
+      })
+      .catch((error) => {
+        this.updateStatus(`There was a problem retrieving the boards: ${error.message}`, 'error');
+      });
+  }
+
+  onBoardSelectionChange = (event) => {
+    let updatedBoardName = {
+      boardName: event.target.value
+    };
+    this.setState(updatedBoardName);
```

Comment:
I like the decision to keep track of which board you're viewing in the `App` component. Trying to force it in elsewhere would have felt clunky.